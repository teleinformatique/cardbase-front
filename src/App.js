import React from "react"
import {AppBar, Typography, Toolbar} from '@material-ui/core';
import Login from './components/Login';

const App = () => {

  return(
    <div className="App">
      <AppBar color="default" position="static">
        <Toolbar>
          <Typography variant="h6" color="inherit">Cardatabase List</Typography>
        </Toolbar>
      </AppBar>
      <Login />
    </div>
    
  )
}

export default App