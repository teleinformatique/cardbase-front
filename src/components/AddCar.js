import  React, {useState}  from "react";
import { Dialog, DialogActions, DialogContent, DialogTitle, Button,TextField } from "@material-ui/core";

const AddCar = (props) =>{
    const [open, setOpen] = useState(false)
    const [car, setCar] = useState({brand: '', model:'', colo:'', price:'', year:''})

    const toggleModal = () =>{
        setOpen(!open)
    }

    const handleChange = (eve) => {
        setCar({...car, [eve.target.name]: eve.target.value})
    }

    const handleSave = () => {
        props.addCar(car)
        toggleModal()
    }

    return(
        <div className='test'>
            <Button variant="outlined" color="primary" onClick={toggleModal} style={{margin:10}}>Add Car</Button>
            <Dialog open={open}>
                <DialogTitle>New Car</DialogTitle>
                <DialogContent>
                    <TextField  label="Brand" name="brand" onChange={handleChange}/> <br/>
                    <TextField  label="Model" name="model" onChange={handleChange}/><br/>
                    <TextField  label="Color" name="color" onChange={handleChange}/><br/>
                    <TextField  label="Year" name="year" onChange={handleChange}/><br/>
                    <TextField label="Price" name="price" onChange={handleChange}/>
                </DialogContent>
                <DialogActions>
                    <Button color="secondary" onClick={toggleModal}>Cancel</Button>
                    <Button color="primary" onClick={handleSave}>Save</Button>
                </DialogActions>
            </Dialog>
        </div>
    )
}

export default AddCar