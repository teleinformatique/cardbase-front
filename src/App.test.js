import { render, screen } from '@testing-library/react';
import App from './App';
import TestRenderer from 'react-test-renderer';
import AddCar from './components/AddCar';

test('renders learn react link', () => {
  render(<App />);
  const linkElement = screen.getByText(/Cardatabase/i);
  expect(linkElement).toBeInTheDocument();
});

it('Render a snapshot', () => {
  const tree = TestRenderer.create(<AddCar />).toJSON()
  expect(tree).toMatchSnapshot()
})