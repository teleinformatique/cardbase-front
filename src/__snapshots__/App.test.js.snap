// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render a snapshot 1`] = `
<div
  className="test"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "margin": 10,
      }
    }
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Add Car
    </span>
  </button>
</div>
`;
